### PR TITLE
docs(orders): ORDERS-4621 Simplify consignments payload in /v2/orders

### PR DESCRIPTION
**Pull request by donald-nguyen-bc**
## What?
This is the doc change for those changes made in these 2 PRs: [#1](https://github.com/bigcommerce/bigcommerce/pull/45470) and [#2](https://github.com/bigcommerce/bigcommerce/pull/45512).

**1. Simplify the response payload of GET requests:**

**1.1. simplify the response payload of `GET /v2/orders/{order_id}/consignments`**

Before change:
```
[
    {
        "id": 2,
        "line_items": {
            "url": "https://api-proxy.service.bcdev/stores/acku7y1wo0/v2/orders/101/products",
            "resource": "/orders/101/products"
        },
        "fulfillment_method": {
            "pickup": {
                "pickup_method_id": 1,
                "pickup_method_display_name": "Pickup Method 1.1",
                "collection_instructions": "Pickup Method 1.1 instructions",
                "collection_time_description": "Pickup Method 1.1 time",
                "location": {
                    "id": 1,
                    "name": "Location 1",
                    "code": "BC-LOCATION-1",
                    "address_line_1": "2801 Skyway Cir",
                    "address_line_2": "",
                    "city": "Austin",
                    "state": "TX",
                    "postal_code": "78704",
                    "country_alpha2": "US",
                    "email": "location1@bigcommerce.com",
                    "phone": "410123001"
                }
            },
            "digital": null,
            "delivery": null
        }
    }
]
```

After change:

```
{
    "pickups": [
        {
            "id": 2,
            "pickup_method_id": 1,
            "pickup_method_display_name": "Pickup Method 1.1",
            "collection_instructions": "Pickup Method 1.1 instructions",
            "collection_time_description": "Pickup Method 1.1 time",
            "location": {
                "id": 1,
                "name": "Location 1",
                "code": "BC-LOCATION-1",
                "address_line_1": "2801 Skyway Cir",
                "address_line_2": "",
                "city": "Austin",
                "state": "TX",
                "postal_code": "78704",
                "country_alpha2": "US",
                "email": "location1@bigcommerce.com",
                "phone": "410123001"
            },
            "line_items": [
                {
                    "url": "https://api-proxy.service.bcdev/stores/acku7y1wo0/v2/orders/101/products/2",
                    "resource": "/orders/101/products/2"
                },
                {
                    "url": "https://api-proxy.service.bcdev/stores/acku7y1wo0/v2/orders/101/products/3",
                    "resource": "/orders/101/products/3"
                }
            ]
        }
    ]
}
```

**1.2. simplify the response payload of `GET /v2/orders/{order_id}/consignments/{consignment_id}`**

Before change:
```
{
    "id": 2,
    "line_items": {
        "url": "https://api-proxy.service.bcdev/stores/acku7y1wo0/v2/orders/101/products",
        "resource": "/orders/101/products"
    },
    "fulfillment_method": {
        "pickup": {
            "pickup_method_id": 1,
            "pickup_method_display_name": "Pickup Method 1.1",
            "collection_instructions": "Pickup Method 1.1 instructions",
            "collection_time_description": "Pickup Method 1.1 time",
            "location": {
                "id": 1,
                "name": "Location 1",
                "code": "BC-LOCATION-1",
                "address_line_1": "2801 Skyway Cir",
                "address_line_2": "",
                "city": "Austin",
                "state": "TX",
                "postal_code": "78704",
                "country_alpha2": "US",
                "email": "location1@bigcommerce.com",
                "phone": "410123001"
            }
        },
        "digital": null,
        "delivery": null
    }
}
```

After change:
```
{
    "id": 2,
    "pickup_method_id": 1,
    "pickup_method_display_name": "Pickup Method 1.1",
    "collection_instructions": "Pickup Method 1.1 instructions",
    "collection_time_description": "Pickup Method 1.1 time",
    "location": {
        "id": 1,
        "name": "Location 1",
        "code": "BC-LOCATION-1",
        "address_line_1": "2801 Skyway Cir",
        "address_line_2": "",
        "city": "Austin",
        "state": "TX",
        "postal_code": "78704",
        "country_alpha2": "US",
        "email": "location1@bigcommerce.com",
        "phone": "410123001"
    },
    "line_items": [
        {
            "url": "https://api-proxy.service.bcdev/stores/acku7y1wo0/v2/orders/101/products/2",
            "resource": "/orders/101/products/2"
        },
        {
            "url": "https://api-proxy.service.bcdev/stores/acku7y1wo0/v2/orders/101/products/3",
            "resource": "/orders/101/products/3"
        }
    ]
}
```

**2. Simplify the request payload of PUT/POST requests:** 

**2.1. simplify the request payload of `POST /v2/orders`** to create an order with pickup consignment: 

Before change:
```
{
    "consignments": [
        {
            "line_items": [
                {
                    "product_id": 104,
                    "quantity": 3,
                    "price_inc_tax": 22.5,
                    "price_ex_tax": 20,
                    "upc": "ABC-123"
                }
            ],
            "fulfillment_method": {
                "delivery": null,
                "digital": null,
                "pickup": {
                    "pickup_method_id": 1,
                    "pickup_method_display_name": "updated Pickup Method 1 name",
                    "collection_instructions": "updated Pickup Method 1 instructions",
                    "collection_time_description": "updated Pickup Method 1 time",
                    "location": {
                        "name": "updated location1",
                        "address_line_1": "updated 2801 Skyway Cir",
                        "address_line_2": "",
                        "city": "Austin",
                        "state": "TX",
                        "postal_code": "78704",
                        "country_alpha2": "US",
                        "email": "updated_location1@bigcommerce.com",
                        "phone": "410123002"
                    }
                }
            }
        }
    ],
    ...
}
```

After change:

```
{
    "consignments": {
        "pickups": [
            {
                "pickup_method_id": 1,
                "pickup_method_display_name": "updated Pickup Method 1 name",
                "collection_instructions": "updated Pickup Method 1 instructions",
                "collection_time_description": "updated Pickup Method 1 time",
                "location": {
                    "name": "updated location1",
                    "address_line_1": "updated 2801 Skyway Cir",
                    "address_line_2": "",
                    "city": "Austin",
                    "state": "TX",
                    "postal_code": "78704",
                    "country_alpha2": "US",
                    "email": "updated_location1@bigcommerce.com",
                    "phone": "410123002"
                },
                "line_items": [
                    {
                        "product_id": 104,
                        "quantity": 3,
                        "price_inc_tax": 22.5,
                        "price_ex_tax": 20,
                        "upc": "ABC-123"
                    }
                ]
            }
        ]
    },
    ....
}
```

**2.2. simplify the request payload of `PUT /v2/orders/{order_id}`** to update consignment info: 

Before change:
```
{
    "consignments": [
        {
            "id": 1,
            "fulfillment_method": {
                "delivery": null,
                "digital": null,
                "pickup": {
                    "pickup_method_id": 2,
                    "pickup_method_display_name": "updated Pickup Method 2 name",
                    "collection_instructions": "updated Pickup Method 2 instructions",
                    "collection_time_description": "updated Pickup Method 2 time",
                    "location": {
                        "name": "updated location 2",
                        "address_line_1": "updated 2802 Skyway Cir",
                        "address_line_2": "",
                        "city": "Austin",
                        "state": "TX",
                        "postal_code": "78704",
                        "country_alpha2": "US",
                        "email": "updated_location2@bigcommerce.com",
                        "phone": "410123002"
                    }
                }
            }
        }
    ],
    ...
}
```

After change:
```
{
    "consignments": {
        "pickups": [
            {
                "id": 1,
                "pickup_method_id": 2,
                "pickup_method_display_name": "updated Pickup Method 2 name",
                "collection_instructions": "updated Pickup Method 2 instructions",
                "collection_time_description": "updated Pickup Method 2 time",
                "location": {
                    "name": "updated location 2",
                    "address_line_1": "updated 2802 Skyway Cir",
                    "address_line_2": "",
                    "city": "Austin",
                    "state": "TX",
                    "postal_code": "78704",
                    "country_alpha2": "US",
                    "email": "updated_location2@bigcommerce.com",
                    "phone": "410123002"
                }
            }
        ]
    },
    ....
}
```

## Why?
To be consistent with storefront order api response, also the new structure is easier to comprehend and consume.

The long term vision for the response payload for GET will be:
```
{
    "pickups": [
      {
        ...
      },
    ],
    "digital": [
      {
        ...
      },
    ],
    "shipping": [
      {
        ...
      },
    ]
}
```

The long term vision for the payload in POST/PUT will be:
```
{
  "consignments": {
    "pickups": [
      {
        ...
      },
    ],
    "digital": [
      {
        ...
      },
    ],
    "shipping": [
      {
        ...
      },
    ]
},
```

## Testing / Proof

The file loaded correctly in Stoplight Studio.

- PUT request payload schema: 

image

- POST request payload schema: 

image

- GET response payload schema: 

image

ping @bigcommerce/orders @traci707

### CODE REVIEW

File: reference/orders.v2.oas2.yml [hunk 1]
```diff
@@ -421,19 +419,27 @@ paths:
...
-                        delivery: null
-                        pickup:
-                          pickup_method_id: 5
-                          pickup_method_display_name: Pick Up - overwrite
-                          location:
-                            name: Location 1 - overwrite
+                    pickups:
+                      - pickup_method_id: 2
+                        pickup_method_display_name: Pick Up - override
+                        collection_instructions: Bring your ID - override.
```
Comment (bc-tgomez) [inline reference/orders.v2.oas2.yml:425]:
```suggestion
                        collection_instructions: Bring your ID - override
```

File: reference/orders.v2.oas2.yml [hunk 2]
```diff
@@ -4690,98 +4689,357 @@ components:
...
+            street_2: string
+            city: Austin
+            state: TX
+            zip: '12345'
+            country: United States
+            country_iso2: US
+            phone: string
+            email: janedoe@example.com
+            form_fields:
+              - name: License Id
```
Comment (bc-tgomez) [inline reference/orders.v2.oas2.yml:4737]:
```suggestion
              - name: License ID
```

File: reference/orders.v2.oas2.yml [hunk 3]
```diff
@@ -4690,98 +4689,357 @@ components:
...
```
Comment (bc-tgomez) [inline reference/orders.v2.oas2.yml:4819]:
```suggestion
              - name: License ID
```

File: reference/orders.v2.oas2.yml [hunk 4]
```diff
@@ -4690,98 +4689,357 @@ components:
...
```
Comment (bc-tgomez) [inline reference/orders.v2.oas2.yml:4917]:
```suggestion
              - name: License ID
```

File: reference/orders.v2.oas2.yml [hunk 5]
```diff
@@ -4690,98 +4689,357 @@ components:
...
```
Comment (bc-tgomez) [inline reference/orders.v2.oas2.yml:4997]:
```suggestion
              - name: License ID
```

File: reference/orders.v2.oas2.yml [hunk 6]
```diff
@@ -4796,108 +5054,371 @@ components:
...
-          type: object
-          description: The pickup location information.
+    pickupConsignment_Post:
+      allOf:
+        - type: object
+          properties:
+            pickup_method_id:
+              type: integer
+              example: 1
+              description: id of the pickup method
```
Comment (bc-tgomez) [inline reference/orders.v2.oas2.yml:5064]:
```suggestion
              description: ID of the pickup method.
```

File: reference/orders.v2.oas2.yml [hunk 7]
```diff
@@ -4796,108 +5054,371 @@ components:
...
-              description: Pickup location's email address.
-              maxLength: 255
-              example: location1@example.com
-            phone:
-              type: string
-              description: Pickup location's phone number.
-              maxLength: 125
-              example: +1 111-111-1111
+              example: 99
+              description: id of the pickup consignment to update
```
Comment (bc-tgomez) [inline reference/orders.v2.oas2.yml:5112]:
```suggestion
              description: ID of the pickup consignment to update.
```

File: reference/orders.v2.oas2.yml [hunk 8]
```diff
@@ -4796,108 +5054,371 @@ components:
...
+            pickup_method_id:
+              type: integer
+              example: 1
+              description: id of the pickup method
```
Comment (bc-tgomez) [inline reference/orders.v2.oas2.yml:5116]:
```suggestion
              description: ID of the pickup method.
```

File: reference/orders.v2.oas2.yml [hunk 9]
```diff
@@ -4796,108 +5054,371 @@ components:
...
+              upc: string
+              id: 12
+    pickupConsignment_Get:
+      allOf:
+        - type: object
+          properties:
+            id:
+              type: integer
+              example: 99
+              description: id of the pickup consignment to update
```
Comment (bc-tgomez) [inline reference/orders.v2.oas2.yml:5163]:
```suggestion
              description: ID of the pickup consignment to update.
```

File: reference/orders.v2.oas2.yml [hunk 10]
```diff
@@ -4796,108 +5054,371 @@ components:
...
+            pickup_method_id:
+              type: integer
+              example: 1
+              description: id of the pickup method
```
Comment (bc-tgomez) [inline reference/orders.v2.oas2.yml:5167]:
```suggestion
              description: ID of the pickup method.
```

File: reference/orders.v2.oas2.yml [hunk 11]
```diff
@@ -4796,108 +5054,371 @@ components:
...
+          country_alpha2: US
+          email: location1@example.com
+          phone: +1 111-111-1111
+      allOf:
+        - type: object
+          properties:
+            id:
+              type: integer
+              example: 1
+              description: id of the location
```
Comment (bc-tgomez) [inline reference/orders.v2.oas2.yml:5292]:
```suggestion
              description: ID of the location.
```

**Review by bc-tgomez — COMMENTED** [review]
Found small items that need a second look, other than that looks good to me! 👍

File: reference/orders.v2.oas2.yml (hunk 1 above)
Comment (donald-nguyen-bc) [reply reference/orders.v2.oas2.yml]: Good catch Tina, updated: https://github.com/bigcommerce/api-specs/pull/676/commits/020f20f5715b0fd0561a2094c54dd8b66dcbbe0e

**Review by WillemHoman — APPROVED** [review]
👏 👏

**Comment by donald-nguyen-bc** [comment]
Hi @traci707, @bc-tgomez, thanks for reviewing this. Since I don't have permission to merge this PR, may you pls do a favour of merging it. Thank you very much in advances.

File: reference/orders.v2.oas2.yml [hunk 12]
```diff
@@ -4690,98 +4689,357 @@ components:
...
-                Consignments field can not be used in conjunction with the products or the shipping_addresses fields.
-
-                Currently only single consignment orders are supported.
-              minItems: 1
-              maxItems: 1
-              items:
-                $ref: '#/components/schemas/orderConsignment_Put'
+              $ref: '#/components/schemas/orderConsignment_Put'
         - $ref: '#/components/schemas/order_Shared'
+      x-examples:
```
Comment (bc-vincent-zhao) [inline reference/orders.v2.oas2.yml:4694]: 👏 more examples

**Pull request merged**